File: 프로그래머스/Level_1/pro_12982.java
// Lv.1 - 예산
import java.util.Arrays;

public class pro_12982 {
    public int solution(int[] d, int budget) {
        int answer = 0;

        Arrays.sort(d);
        for(int i: d){
            budget -= i;
            if(budget < 0) break;
            else answer++;
        }

        return answer;
    }

    public static void main(String[] args) {
        pro_12982 b = new pro_12982();
        int[] d1 = {1, 3, 2, 5, 4};
        int[] d2 = {2, 2, 3, 3};

        System.out.println(b.solution(d1, 9));
        System.out.println(b.solution(d2, 10));
    }
}

File: 프로그래머스/Level_1/pro_12982.py
# 예산


def solution(d, budget):
    answer = 0
    d.sort()

    for x in d:
        budget -= x
        if budget < 0:
            break
        else:
            answer += 1

    return answer


arr = list(map(int, input().split()))
budget = int(input())
print(solution(arr, budget))
